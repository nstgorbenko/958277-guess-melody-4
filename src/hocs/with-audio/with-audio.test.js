import PropTypes from "prop-types";
import React from "react";
import renderer from "react-test-renderer";

import withAudio from "./with-audio.js";

const MockComponent = (props) => {
  return (
    <div>
      {props.children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAudio(MockComponent);

describe(`WithAudio HOC rendering`, () => {
  it(`WithAudio should render correctly`, () => {
    const tree = renderer
      .create(<MockComponentWrapped
        isPlaying={false}
        onPlayButtonClick={() => {}}
        src={``}
      />, {
        createNodeMock: () => ({})
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
