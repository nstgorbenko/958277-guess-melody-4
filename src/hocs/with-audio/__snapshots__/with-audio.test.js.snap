// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithAudio HOC rendering renders correctly 1`] = `
<div>
  <audio />
</div>
`;
